app/src/main/java/com/patrickgatewood/weather/ui/WeatherPresenterImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.patrickgatewood.weather.ui;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.patrickgatewood.weather.data.model.local.Constants;
+import com.patrickgatewood.weather.data.model.remote.request.DarkSkyApi;
+import com.patrickgatewood.weather.data.model.remote.response.Forecast;
+import com.patrickgatewood.weather.data.model.remote.response.ForecastData;
+
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class WeatherPresenterImpl implements WeatherPresenter {
+
+    private static final String TAG = "WeatherPresenter";
+
+    @NonNull
+    private DarkSkyApi darkSkyApi;
+
+    @Nullable
+    private Forecast currentForecast = null;
+
+    @Nullable
+    private WeatherView weatherView;
+
+    @Inject
+    public WeatherPresenterImpl(@NonNull DarkSkyApi darkSkyApi) {
+        this.darkSkyApi = darkSkyApi;
+    }
+
+    public void onFetchButtonClick() {
+        Log.v(TAG, "API call initiated");
+
+        // TODO get user's location and pass to query
+        queryApi();
+    }
+
+    @Override
+    public void attachView(Context weatherContext) {
```

Comment:
@nishtahir Since it'd be safer, I'll have the interface use generics. That way there's less type checking and casting